privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -5,61 +5,49 @@
 from privacy_evaluator.utils.trainer import trainer
 from privacy_evaluator.models.torch.fc_neural_net import FCNeuralNet
 
-import math
 import numpy as np
 import torch
 import tensorflow as tf
 from sklearn.svm import SVC
 from sklearn.model_selection import train_test_split
+from sklearn.utils import shuffle
 from art.attacks.evasion import FastGradientMethod
 from art.estimators.classification import SklearnClassifier
-from typing import Tuple, Any, Dict, List
+from typing import Tuple, Any, Dict, List, Optional
 
 
 class PropertyInferenceAttack(Attack):
-    def __init__(self, target_model: Classifier, dataset: Tuple[np.ndarray,np.ndarray]):
+    def __init__(self, target_model: Classifier, dataset: Tuple[np.ndarray, np.ndarray]):
         """
         Initialize the Property Inference Attack Class.
         :param target_model: the target model to be attacked
         :param dataset: dataset for training of shadow classifiers, test_data from dataset with concatenation [test_features, test_labels]
         """
-        self.dataset=dataset
-        # count of shadow training sets, must be eval
-        self.amount_sets = 6
-        self.input_shape =  self.dataset[0].shape #[32, 32, 3]
+        self.dataset = dataset
+        # count of shadow training sets, must be even
+        self.amount_sets = 2 #TODO change to higher number
+        self.input_shape = self.dataset[0][0].shape  # [32, 32, 3] for CIFAR10
         super().__init__(target_model, None, None, None, None)
 
     def create_shadow_training_set(
         self,
         num_elements_per_class: Dict[int, int],
-    ) ->List[Tuple[np.ndarray, np.ndarray]]:
+    ) -> List[Tuple[np.ndarray, np.ndarray]]:
         """
         Create the shadow training sets with given ratio.
         The function works for the specific binary case that the ratio is a fixed distribution specified in the input.
-        :param num_elements_per_classes: number of elements per class
+        :param num_elements_per_class: number of elements per class
         :return: shadow training sets for given ratio
         """
 
         training_sets = []
-        #neg_property_training_sets = []
-
-        """negation_num_elements_per_class = {
-            class_id: int(
-                round(
-                    sum(property_num_elements_per_class.values())
-                    / len(property_num_elements_per_class)
-                )
-            )
-            for class_id in property_num_elements_per_class.keys()
-        }"""
 
         # Creation of shadow training sets with the size dictionaries
-        for i in range(int(self.amount_sets / 2)):
+        for _ in range(int(self.amount_sets / 2)):
```

Comment:
please add comment why you divide by two

Reply:
I added a comment with an explanation.